reverse_image_search_module/README.md

```diff
@@ -0,0 +1,105 @@
+
+# Image Retrieval using ResNet-18 and Annoy Index
+
+This folder contains code for performing reverse image search using the ResNet-18 neural network for feature extraction and the Annoy Index for efficient nearest neighbor search. The provided code allows you to generate embeddings for a collection of images, build an Annoy Index from those embeddings, and perform reverse image searches to find similar images from a dataset.
+
+
+# Image Retrieval using ResNet-18 and Annoy Index
+
+This repository contains code for performing reverse image search using the ResNet-18 neural network for feature extraction and the Annoy Index for efficient nearest neighbor search.
+
+## Dependencies
+
+To use the code in this repository, you will need the following dependencies:
+
+- Python 3.x
+- PyTorch
+- torchvision
+- numpy
+- pandas
+- matplotlib
+- annoy
+
+You can install these dependencies using the following command:
+
+```
+pip install torch torchvision numpy pandas matplotlib annoy
+```
+## Usage
+
+Feature Extraction with ResNet-18
+
+The Img2VecResnet18 class provides methods to extract image embeddings using the ResNet-18 neural network.
+
+- getVec(img): Extracts the raw image embedding for a given image.
+- getNormalizedVec(img): Extracts the normalized image embedding for a given image.
+
+Example usage:
+
+```
+from resnet_image_search import Img2VecResnet18
+import numpy as np
+import matplotlib.pyplot as plt
+import torchvision.transforms as transforms
+
+# Initialize the Img2VecResnet18 instance
+img2vec = Img2VecResnet18()
+
+# Load an image and get its normalized embedding
+image = plt.imread('path_to_your_image.jpg')
+normalized_embedding = img2vec.getNormalizedVec(image)
+```
+
+## Building the Annoy Index
+
+The Annoy Index is built using embeddings from a collection of images.
```

Comment:
You can say couple of words about why annoy (it's quite efficient and easy to use - maybe even easier than sklearn or on the same level)